test/integration/api/forms.js

```diff
@@ -1380,6 +1380,137 @@ describe('api: /projects/:id/forms', () => {
+            .expect(200)
+            .then(({ body }) => {
+              should.exist(body[0].deletedAt);
+              body.forEach((form) => form.should.be.a.Form());
+              body.map((form) => form.id).should.eql([ 2 ]);
+              body.map((form) => form.projectId).should.eql([ 1 ]);
+              body.map((form) => form.xmlFormId).should.eql([ 'withrepeat' ]);
+              body.map((form) => form.name).should.eql([ null ]);
+              body.map((form) => form.hash).should.eql([ 'e7e9e6b3f11fca713ff09742f4312029' ]);
+              body.map((form) => form.version).should.eql([ '1.0' ]);
+            })))));
+
+    it('should list soft-deleted forms including extended metadata and submissions', testService((service) =>
+      service.login('alice', (asAlice) =>
+         asAlice.post('/v1/projects/1/forms/simple/submissions')
+          .send(testData.instances.simple.one)
+          .set('Content-Type', 'application/xml')
+          .expect(200)
+          .then(() => asAlice.delete('/v1/projects/1/forms/simple')
+            .expect(200)
+            .then(() => asAlice.get('/v1/projects/1/forms?deleted=true')
+              .set('X-Extended-Metadata', 'true')
+              .expect(200)
+              .then(({ body }) => {
+                const simple = body.find((form) => form.xmlFormId === 'simple');
+                should.exist(simple.deletedAt);
+                simple.submissions.should.equal(1);
+                simple.lastSubmission.should.be.a.recentIsoDate();
+              }))))));
+
+    it('should not include deletedAt form field on regular forms', testService((service) =>
+      service.login('alice', (asAlice) =>
+        asAlice.get('/v1/projects/1/forms')
+          .set('X-Extended-Metadata', 'true')
+          .expect(200)
+          .then(({ body }) => {
+            body.forEach((form) => form.hasOwnProperty('deletedAt').should.be.false());
+          })
+          .then(() => asAlice.get('/v1/projects/1/forms')
+            .expect(200)
+            .then(({ body }) => {
+              body.forEach((form) => form.hasOwnProperty('deletedAt').should.be.false());
+            })))));
+  });
+
+  describe('/:id/restore (undeleting trashed forms)', () => {
+    it('should reject restoring unless the user can delete', testService((service) =>
+      service.login('alice', (asAlice) =>
+        asAlice.delete('/v1/projects/1/forms/simple')
+          .expect(200)
+          .then(() => service.login('chelsea', (asChelsea) =>
+            asChelsea.post('/v1/projects/1/forms/1/restore').expect(403))))));
+
+    it('should reject restoring if referenced by the xml form id', testService((service) =>
+      service.login('alice', (asAlice) =>
+        asAlice.delete('/v1/projects/1/forms/simple')
+          .expect(200)
+          .then(() => asAlice.post('/v1/projects/1/forms/simple/restore')
+            .expect(400))))); // bad request
```

Comment:
clarify that this is 400.11 `message: "Invalid input data type: expected 'simple' to be (type integer)",`